### PR TITLE
Add deltaTime and world reference to Systems

### PR DESCRIPTION
**Pull request by giacomocavalieri**

### CODE REVIEW

File: core/src/main/scala/dev/atedeg/ecscala/System.scala [hunk 1]
```diff
@@ -34,13 +56,13 @@ trait System[L <: CList] extends ((Entity, L, View[L]) => Deletable[L]) {
    * This method should only be called with a CListTag[L], it can not be enforced in the interface as it would make it
    * impossible to call it correctly from the World requiring to cast the tags to a type that has been erased.
    */
-  private[ecscala] def update(world: World)(using clt: CListTag[? <: CList]): Unit = {
+  private[ecscala] def update(world: World, deltaTime: Float)(using clt: CListTag[? <: CList]): Unit = {
```
Comment (ndido98) [inline core/src/main/scala/dev/atedeg/ecscala/System.scala:59]: You forgot to use the DeltaTime type alias

File: core/src/main/scala/dev/atedeg/ecscala/World.scala [hunk 2]
```diff
@@ -48,7 +48,7 @@ sealed trait World {
    */
   def addSystem[L <: CList](system: System[L])(using ct: CListTag[L]): Unit
 
-  def update(): Unit
+  def update(deltaTime: Float): Unit
```
Comment (ndido98) [inline core/src/main/scala/dev/atedeg/ecscala/World.scala:51]: Here too

File: core/src/main/scala/dev/atedeg/ecscala/World.scala [hunk 3]
```diff
@@ -88,9 +88,9 @@ object World {
     override def addSystem[L <: CList](system: System[L])(using ct: CListTag[L]): Unit =
       systems = systems :+ (ct -> system)
 
-    override def update(): Unit = systems foreach (taggedSystem => {
+    override def update(deltaTime: Float): Unit = systems foreach (taggedSystem => {
```
Comment (ndido98) [inline core/src/main/scala/dev/atedeg/ecscala/World.scala:91]: Here too

File: core/src/test/scala/dev/atedeg/ecscala/SystemTest.scala [hunk 4]
```diff
@@ -44,36 +42,58 @@ class SystemTest extends AnyWordSpec with Matchers {
...
+              val (entity, Position(px, py) &: _) = entityComponentsPair
+              entity.addComponent(Position(px + 1, py + 1))
+            })
 
-        override def after(world: World, view: View[Comps]) =
-          view foreach (entityComponentsPair => {
-            val (entity, Position(px, py) &: _) = entityComponentsPair
-            entity.addComponent(Position(px + 1, py + 1))
-          })
+          override def apply(entity: Entity, components: Comps, deltaTime: Float, world: World, view: View[Comps]) =
```
Comment (ndido98) [inline core/src/test/scala/dev/atedeg/ecscala/SystemTest.scala:85]: Here too?

**Review by ndido98 — CHANGES_REQUESTED** [review]

**Review by ndido98 — APPROVED** [review]

**Review by nicolasfara — APPROVED** [review]

**Review by vitlinda — APPROVED** [review]

**Pull request merged**